tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/GraphOperation.java

```diff
@@ -147,25 +162,92 @@ Shape shape(int outputIdx) {
     throw new IllegalStateException("Graph tensors must be fetched by running a session");
   }
 
-  long getUnsafeNativeHandle() {
+  TF_Operation getUnsafeNativeHandle() {
     return unsafeNativeHandle;
   }
 
   private final Graph graph;
 
-  private final long unsafeNativeHandle;
+  private final TF_Operation unsafeNativeHandle;
 
-  private static native String name(long handle);
+  private static void requireHandle(Pointer handle) {
+    if (handle == null || handle.isNull()) {
+      throw new IllegalStateException("close() has been called on the Graph this Operation was a part of");
+    }
+  }
+
+  private static String name(TF_Operation handle) {
+    requireHandle(handle);
+    return TF_OperationName(handle).getString();
```

Comment:
`BytePointer` returned by these methods don't need to be freed/scoped?

Reply:
No, as explained above, by convention, JavaCPP doesn't perform any allocation for directly mapped C/C++ functions.